src/java.base/share/classes/java/nio/file/Files.java

```diff
@@ -1525,29 +1513,19 @@ private static class FileTypeDetectors{
             loadInstalledDetectors();
 
         // creates the default file type detector
-        @SuppressWarnings("removal")
         private static FileTypeDetector createDefaultFileTypeDetector() {
-            return AccessController
-                .doPrivileged(new PrivilegedAction<>() {
-                    @Override public FileTypeDetector run() {
-                        return sun.nio.fs.DefaultFileTypeDetector.create();
-                }});
+            return sun.nio.fs.DefaultFileTypeDetector.create();
         }
 
         // loads all installed file type detectors
-        @SuppressWarnings("removal")
         private static List<FileTypeDetector> loadInstalledDetectors() {
-            return AccessController
-                .doPrivileged(new PrivilegedAction<>() {
-                    @Override public List<FileTypeDetector> run() {
-                        List<FileTypeDetector> list = new ArrayList<>();
-                        ServiceLoader<FileTypeDetector> loader = ServiceLoader
-                            .load(FileTypeDetector.class, ClassLoader.getSystemClassLoader());
-                        for (FileTypeDetector detector: loader) {
-                            list.add(detector);
-                        }
-                        return list;
-                }});
+            List<FileTypeDetector> list = new ArrayList<>();
+            ServiceLoader<FileTypeDetector> loader = ServiceLoader
+                .load(FileTypeDetector.class, ClassLoader.getSystemClassLoader());
+            for (FileTypeDetector detector: loader) {
+                list.add(detector);
+            }
+            return list;
```

Comment:
Possible refactor to use stream API:  (may be a bit too aggressive)
```suggestion
             return ServiceLoader
                    .load(FileTypeDetector.class, ClassLoader.getSystemClassLoader())
                    .stream()
                    .map(p -> (FileTypeDetector)p).toList();
```

Reply:
Addressed in #22552.